eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -86,53 +86,93 @@ stages:
                     steps:
                       - checkout: self
 
+                      - pwsh: |
+                          $branchName = "azure-sdk-for-cpp-${{ artifact.Name }}-$(Build.BuildId)"
+                          if ($env:VCPKGFORKBRANCHNAME) { 
+                            Write-Host "Using queue time branch name" 
+                            $branchName = $env:VCPKGFORKBRANCHNAME
+                          } 
+                          Write-Host "##vso[task.setvariable variable=PrBranchName]$branchName"
+                        displayName: Set fork branch name
+
                       - task: Powershell@2
                         inputs:
                           filePath: eng/scripts/Initialize-VcpkgRelease.ps1
                           arguments: >-
-                            -GitHubPat $(azuresdk-github-pat)
                             -GitHubRepo "$(Build.Repository.Name)"
                             -PackageSpecPath $(Pipeline.Workspace)/packages/${{artifact.Name}}/package-info.json
                             -SourceDirectory $(Pipeline.Workspace)/packages/${{artifact.Name}}/vcpkg
                             -Verbose
                           pwsh: true
                         displayName: Initialize vcpkg release
 
+                      # There are potential race conditions if this script runs
+                      # in parallel against the same branch name. Release only 
+                      # one package at a time.
                       - pwsh: |
+                          Write-Host "git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg"
                           git clone https://github.com/microsoft/vcpkg $(Pipeline.Workspace)/vcpkg
                           if ($LASTEXITCODE -ne 0) {
                             Write-Error "Unable to check out vcpkg repo"
-                            exit $LASTEXITCODE
+                            exit $LASTEXITCODE`
                           }
+                          
+                          Write-Host "cd $(Pipeline.Workspace)/vcpkg"
                           cd $(Pipeline.Workspace)/vcpkg
-
-                          # There is no need to check out a branch, the
-                          # create-pull-request.yml template creates the branch
+                          Write-Host "git remote add fork https://github.com/azure-sdk/vcpkg -f"
+                          git remote add fork https://github.com/azure-sdk/vcpkg -f
+
+                          # If there is a remote branch by this name check it 
+                          # out. ls-remote with --exit-code will exit with a 
+                          # status code of 2 indicates that the remote branch 
+                          # does not exist
+                          # https://git-scm.com/docs/git-ls-remote.html
+                          Write-Host "git ls-remote --exit-code --heads fork $(PrBranchName)"
```

Comment:
I wouldn't worry trying to find any remote branches I'd just create a temp branch do the changes and then call create-pull-request and let it do the rest.